### PR TITLE
Fix for root update

### PR DESCRIPTION
**Pull request by guruchandru**

### CODE REVIEW

File: src/webcfg_multipart.c [hunk 1]
```diff
@@ -658,7 +658,14 @@ WEBCFG_STATUS processMsgpackSubdoc(char *transaction_id)
...
 							addWebConfgNotifyMsg(mp->entries[m].name_space, mp->entries[m].etag, "failed", result, trans_id,0,"status",ccspStatus, NULL, 200);
-							WebcfgDebug("the retry flag value is %d\n", get_doc_fail());
+							WebcfgInfo("checkRootUpdate\n");
+							if(checkRootUpdate() == WEBCFG_SUCCESS)
+							{
+								WebcfgInfo("updateRootVersionToDB\n");
+								updateRootVersionToDB();
+							}
+							addNewDocEntry(get_successDocCount());
+							WebcfgInfo("the retry flag value is %d\n", get_doc_fail());
```
Comment (sadhyama) [inline src/webcfg_multipart.c:668]: Once all docs are success and root is updated, make sure it is not going to for loop to process again. So in this 204 case add break to for loop once root is updated and mp cache is freed.

**Review by sadhyama — CHANGES_REQUESTED** [review]

File: src/webcfg_multipart.c [hunk 2]
```diff
@@ -658,7 +658,21 @@ WEBCFG_STATUS processMsgpackSubdoc(char *transaction_id)
 							WebcfgDebug("The result is %s\n",result);
 							updateTmpList(subdoc_node, mp->entries[m].name_space, mp->entries[m].etag, "failed", result, ccspStatus, 0, 1);
 							addWebConfgNotifyMsg(mp->entries[m].name_space, mp->entries[m].etag, "failed", result, trans_id,0,"status",ccspStatus, NULL, 200);
-							WebcfgDebug("the retry flag value is %d\n", get_doc_fail());
+							WebcfgInfo("checkRootUpdate\n");
+							if(checkRootUpdate() == WEBCFG_SUCCESS)
```
Comment (shilpa24balaji) [inline src/webcfg_multipart.c:662]: Lets move this root check inside the 204 case or at the end of multipart processing after aker

Comment (shilpa24balaji) [reply src/webcfg_multipart.c:662]: You can add check (ccspStatus == 204 && subdocStatus != WEBCFG_SUCCESS) also here so its only for unsupported case

File: src/webcfg_multipart.c [hunk 3]
```diff
@@ -658,7 +658,21 @@ WEBCFG_STATUS processMsgpackSubdoc(char *transaction_id)
 							WebcfgDebug("The result is %s\n",result);
 							updateTmpList(subdoc_node, mp->entries[m].name_space, mp->entries[m].etag, "failed", result, ccspStatus, 0, 1);
 							addWebConfgNotifyMsg(mp->entries[m].name_space, mp->entries[m].etag, "failed", result, trans_id,0,"status",ccspStatus, NULL, 200);
-							WebcfgDebug("the retry flag value is %d\n", get_doc_fail());
+							WebcfgInfo("checkRootUpdate\n");
```
Comment (shilpa24balaji) [inline src/webcfg_multipart.c:661]: Remove this log

File: src/webcfg_multipart.c [hunk 4]
```diff
@@ -658,7 +658,21 @@ WEBCFG_STATUS processMsgpackSubdoc(char *transaction_id)
...
+							if((ccspStatus == 204 && subdocStatus != WEBCFG_SUCCESS) && (checkRootUpdate() == WEBCFG_SUCCESS))
+							{
+								WebcfgInfo("updateRootVersionToDB\n");
```
Comment (shilpa24balaji) [inline src/webcfg_multipart.c:664]: Is this info log needed?

**Review by sadhyama — APPROVED** [review]

**Pull request merged**